opensrp-chw/src/main/java/org/smartregister/chw/activity/MalariaProfileActivity.java

```diff
@@ -0,0 +1,131 @@
+import org.smartregister.chw.malaria.domain.MemberObject;
+import org.smartregister.chw.malaria.util.Constants;
+import org.smartregister.commonregistry.CommonPersonObject;
+import org.smartregister.commonregistry.CommonPersonObjectClient;
+import org.smartregister.commonregistry.CommonRepository;
+import org.smartregister.family.util.JsonFormUtils;
+import org.smartregister.family.util.Utils;
+import timber.log.Timber;
+
+public class MalariaProfileActivity extends BaseMalariaProfileActivity {
+    private final String TAG = MalariaProfileActivity.class.getCanonicalName();
+    private static final String CLIENT = "client";
+    private View view;
+    private MemberObject memberObject;
+    private ChildProfileActivityFlv flavor = new ChildProfileActivityFlv();
+
+    public static void startMalariaActivity(Activity activity, MemberObject memberObject, CommonPersonObjectClient client) {
+        Intent intent = new Intent(activity, MalariaProfileActivity.class);
+        intent.putExtra(Constants.MALARIA_MEMBER_OBJECT.MEMBER_OBJECT, memberObject);
+        intent.putExtra(CLIENT, client);
+        activity.startActivity(intent);
+    }
+
+    @Override
+    protected void onCreation() {
+        super.onCreation();
+        memberObject = (MemberObject) getIntent().getSerializableExtra(Constants.MALARIA_MEMBER_OBJECT.MEMBER_OBJECT);
+    }
+
+    @Override
+    public void onClick(View view) {
+        onBackPressed();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.other_member_menu, menu);
+        if (flavor.showMalariaConfirmationMenu()) {
+            menu.findItem(R.id.action_malaria_registration).setVisible(false);
+            menu.findItem(R.id.action_malaria_followup_visit).setVisible(true);
+        }
+        menu.findItem(R.id.action_anc_registration).setVisible(false);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                return true;
+            case R.id.action_registration:
+                startFormForEdit(memberObject.getBaseEntityId(), org.smartregister.chw.util.Constants.JSON_FORM.FAMILY_MEMBER_REGISTER, R.string.registration_info, memberObject.getLastName());
+                return true;
+
+            case R.id.action_malaria_followup_visit:
+                Toast.makeText(getApplicationContext(), "Malaria Follow up", Toast.LENGTH_SHORT).show();
+                return true;
+
+            case R.id.action_remove_member:
+
+                CommonRepository commonRepository = org.smartregister.chw.util.Utils.context().commonrepository(org.smartregister.chw.util.Utils.metadata().familyMemberRegister.tableName);
+
+                final CommonPersonObject commonPersonObject = commonRepository.findByBaseEntityId(memberObject.getBaseEntityId());
```

Comment:
since we are using the CommonPersonObject in multiple places  within this class could we find a way to reuse?